### ISSUE TITLE
[onert-micro] Differentiate `flat_size` and length of `target_data`

### ISSUE THREAD

**Issue by ljwoo94** [Samsung/ONE#13404]
## Background 
- We need to bring up `SparseCrossEntropy` for loss function.

## What
- A `SparseCrossEntropy` is a metric for classification that has `integer indices` as a target_label.
- Current metrics/loss_functions support `CrossEntropy` that assumes a one-hot-encoded target_label.
- (e.g) 
  - CrossEntropy : 
    ``` 
    target_data = [0,1,0]
    calculated_data = [0.1, 0.8, 0.1]
    ```
  - SparseCrossEntropy: 
    ``` 
    target_data = [1]
    calculated_data = [0.1, 0.8, 0.1]
    ```
- In the above case, `flat_size==category_num` can assumed as same number that can represent length of calculated data and target data.
- However, in `SparseCrossEntropy` this flat_size assumption  cannot be maintained and might occurs unexpected memory movement.
  - https://github.com/Samsung/ONE/blob/98ed924fefc8f498d5cf0f049adec1853de0d6cb/onert-micro/onert-micro/src/core/train/OMTrainingHandler.cpp#L54-L63

## Solution(?)
1. **Add `One-Hot-Encoding` feature.**
    - update target_data with this encoded value.
    - call CrossEntropy by using new_encoded_value as a target_data.
2. **Fix memory offset calculation logic for target data.** 

Also, this might not be an issue if I misunderstood the logic.
